### PR TITLE
Issue 94: writeUNodeUri adds unnecessary colons

### PR DESCRIPTION
**Pull request by tmciver**
This is the second part of the fix for the issues described in #94 and #27.

### CODE REVIEW

File: rdf4h.cabal [hunk 1]
```diff
@@ -124,6 +124,8 @@ test-suite test-rdf4h
                , filepath
                , directory
                , safe
+               , temporary
```
Comment (robstewart57) [inline rdf4h.cabal:127]: @tmciver  It seems a slight shame that this introduces an additional dependency, which itself brings in its own dependencies that rdf4h doesn't rely on (random, transformers, unix), though these may be transitive dependencies from rdf4h's existing dependencies.

That said, it's a dependency of the rdf4h test-suite, not of the rdf4h library. So for most users just depending on the rdf4h library won't pull those additional dependencies in via `temporary`.

File: testsuite/tests/Text/RDF/RDF4H/TurtleSerializerTest.hs [hunk 2]
```diff
@@ -2,17 +2,29 @@
...
-tests =
-  testGroup "findMappings Tests"
-  [ testCase "findMapping correctly finds rdf mapping" $
-    assertEqual "" (Just ("http://www.w3.org/1999/02/22-rdf-syntax-ns#", "subject")) (findMapping (coerce standard_ns_mappings) "rdf:subject")
-  , testCase "findMapping correctly finds rdfs mapping" $
-    assertEqual "" (Just ("http://www.w3.org/2000/01/rdf-schema#", "domain")) (findMapping standard_ns_mappings "rdfs:domain")
+tests = testGroup "Turtle serializer tests"
+  [ testGroup "findMappings Tests"
+    [ testCase "findMapping correctly finds rdf mapping" $
+      assertEqual "" (Just ("http://www.w3.org/1999/02/22-rdf-syntax-ns#", "subject")) (findMapping (coerce standard_ns_mappings) "rdf:subject")
```
[the remark this answers is not in the document]
Comment (tmciver) [reply testsuite/tests/Text/RDF/RDF4H/TurtleSerializerTest.hs]: Yes, I forgot that one from the previous PR.  Fixed now.

File: rdf4h.cabal (hunk 1 above)
Comment (tmciver) [reply rdf4h.cabal:127]: Yes, I figured too that since it's in the testsuite, that it wasn't a big deal.

Comment (tmciver) [reply rdf4h.cabal:127]: Also, I wanted to ask about your thoughts on exposing functions from modules.  In PR #95 I had exposed `findMapping` and `writeUNodeUri` which were not previously exported.  It's a shame, IMO, to have to do this; I wish Haskell had a way of only exposing functions to the test package, or something like that.

There are a few options.  First, we could not expose such functions and just test them through the functions that are exported.  Another would be to use often-used technique of having a set of `Internal` modules where everything is exposed and we just rely on convention to keep users from depending on the things exposed there.  Finally, we could just expose more such functions and consider it part of the API.  Thoughts?

Comment (robstewart57) [reply rdf4h.cabal:127]: @tmciver Iet's move this discussion over to https://github.com/robstewart57/rdf4h/pull/95

**Comment by robstewart57** [comment]
@tmciver thanks, merged.

**Pull request merged**